src/app/fdctl/run/tiles/fd_replay.c

```diff
@@ -510,33 +521,52 @@ unprivileged_init( fd_topo_t *      topo,
     FD_LOG_ERR(( "no funk workspace" ));
   }
 
-  fd_funk_t * funk;
-  void * shmem;
-  shmem = fd_wksp_alloc_laddr( ctx->funk_wksp, fd_funk_align(), fd_funk_footprint(), FD_FUNK_MAGIC );
-  if (shmem == NULL)
-    FD_LOG_ERR(( "failed to allocate a funky" ));
-  funk = fd_funk_join( fd_funk_new( shmem, FD_FUNK_MAGIC, ctx->funk_seed, tile->replay.txn_max, tile->replay.index_max ) );
-  if (funk == NULL) {
-    fd_wksp_free_laddr(shmem);
-    FD_LOG_ERR(( "failed to allocate a funky" ));
+  fd_funk_t * funk = NULL;
+  void * shmem = NULL;
+  ctx->snapshot = tile->replay.snapshot;
+  if ( strncmp(ctx->snapshot, "wksp:", 5) == 0 ) {
```

Comment:
I think we should have a separate field for the wksp name. As far as I can tell snapshot is just getting used as either a snapshot file or a wksp file name which is unclean. These checks should become consistent with all the other arg related checks, instead of checking for a prefix.

Reply:
urls will use the http: prefix for snapshot, so I'm just following this convention

Reply:
We need to know the slot number from the snapshot filename. We should probably not be doing HTTP requests this late in start up.

Reply:
Granted the prefix convention might change, but until it does we should have what is current. And even with different prefixes, I don't think this field should be used for both cases. Or it should be renamed to reflect that it can be either